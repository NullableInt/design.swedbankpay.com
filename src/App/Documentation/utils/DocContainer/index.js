import React from "react";
import PropTypes from "prop-types";

import { DocToc } from "#";

const DocContainer = ({ docToc, children }) => {
    const DocContent = () => <div className={`doc-body ${docToc ? "col-lg-10" : "col-12"}`}>{children}</div>;

    return (
        <div className="doc-container">
            <div className="row">
                <DocContent />
                {docToc ? <DocToc component={DocContent} /> : null}
            </div>
        </div>
    );
};

DocContainer.propTypes = {
    docToc: PropTypes.bool
};

export default DocContainer;
