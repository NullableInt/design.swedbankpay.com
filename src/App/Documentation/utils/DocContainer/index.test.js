import React from "react";
import { mount } from "enzyme";

import DocContainer from "./index";

describe("Utilities: DeprecatedComponentAlert", () => {
    it("is defined", () => {
        expect(DocContainer).toBeDefined();
    });

    it("renders", () => {
        const wrapper = mount(<DocContainer>Component</DocContainer>);

        expect(wrapper).toMatchSnapshot();
    });

    it("renders with a table of contents(doctoc) if prop docToc is given", () => {
        const wrapper = mount(<DocContainer docToc>Component</DocContainer>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.html()).toContain("doc-toc");
    });
});
