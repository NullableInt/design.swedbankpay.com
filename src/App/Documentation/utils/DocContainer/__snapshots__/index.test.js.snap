// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities: DeprecatedComponentAlert renders 1`] = `
<DocContainer>
  <div
    className="doc-container"
  >
    <div
      className="row"
    >
      <DocContent>
        <div
          className="doc-body col-12"
        >
          Component
        </div>
      </DocContent>
    </div>
  </div>
</DocContainer>
`;

exports[`Utilities: DeprecatedComponentAlert renders with a table of contents(doctoc) if prop docToc is given 1`] = `
<DocContainer
  docToc={true}
>
  <div
    className="doc-container"
  >
    <div
      className="row"
    >
      <DocContent>
        <div
          className="doc-body col-lg-10"
        >
          Component
        </div>
      </DocContent>
      <DocToc
        component={[Function]}
      >
        <div
          className="col-2 d-none d-lg-block"
        >
          <div
            className="doc-toc"
          >
            <ul />
          </div>
        </div>
      </DocToc>
    </div>
  </div>
</DocContainer>
`;
